nodebuilder/core/config.go

```diff
@@ -0,0 +1,58 @@
+package core
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+// Config combines all configuration fields for managing the relationship with a Core node.
+type Config struct {
+	IP       string
+	RPCPort  string
+	GRPCPort string
+}
+
+// DefaultConfig returns default configuration for managing the
+// node's connection to a Celestia-Core endpoint.
+func DefaultConfig() Config {
+	return Config{
+		IP:       "0.0.0.0",
+		RPCPort:  "0",
+		GRPCPort: "0",
```

Comment:
Yes because both light nodes and full nodes can technically operate without a celestia-core node connection. A problem would only arise if they try to perform state interaction (e.g. requesting their balance via `/balance` or submitting a tx) because there is no celestia-core node connection to proxy the request to.